Dungeoneer/src/com/interrupt/dungeoneer/statuseffects/BurningEffect.java

```diff
@@ -103,6 +103,177 @@ public void onStatusBegin(Actor owner) {
 	public void onStatusEnd(Actor owner) {
 		this.active = false;
 		Fire fire = (Fire)owner.getAttached(Fire.class);
-		if(fire != null) fire.isActive = false;
+		if (fire != null) fire.isActive = false;
 	}
+
+
+    /** Used for putting out the fire */
+	private boolean isEntityInWater() {
+        if (owner.inwater) {
+            Tile waterTile = Game.instance.level.findWaterTile(owner.x, owner.y, owner.z, owner.collision);
+            return waterTile.data != null && (waterTile.data.hurts <= 0 || waterTile.data.damageType != DamageType.FIRE);
+        }
+        return false;
+    }
+
+    /** Constructs new fire particle for spawning */
+    private void spawnFireParticle() {
+
+        Particle p = CachePools.getParticle();
+
+        p.tex = particleTexture;
+        p.lifetime = particleLifetime;
+        p.scale = scale;
+        p.startScale = startScale;
+        p.endScale = endScale;
+        p.fullbrite = true;
+        p.checkCollision = false;
+        p.floating = true;
+        setParticlePosition(p);
+
+        p.playAnimation(startTexture, stopTexture, animationSpeed);
+
+        p.za = Game.rand.nextFloat() * upwardVelocity + upwardVelocity;
```

Comment:
This might be a good time to fix this magic math. 😄

Reply:
Every Status Effect particle goes by a formula:  **za = randFloat() * randFactor + baseFactor**, except for SlowEffect particle, which has **za = const**.

I think that simplifying the formuma to **za = const** will make them look to uniform and bland - with particles overlapping when the entity stands still. 

My proposition here is to keep them as is - extract the hardcoded values and keep them as a public fields.
They should be named as something along the lines of **baseZAFactor / baseZA** and **randZAFactor / randZA**
to be consistent with the **xa, ya, za** fields of the Particle class.

My mistake here was that i've extracted the hardcoded value, but i've forgot to split it to **baseFactor** and **randFactor** (because it was the same in both cases (0.004f)).